### ISSUE TITLE
`null` return typehint generated when types conflict

### ISSUE THREAD

**Issue by othercorey** [slevomat/coding-standard#1241]
It looks like `SlevomatCodingStandard.TypeHints.ReturnTypeHint` is generating `null` return types when the type specified by `@return` annotation and the actual return statement conflict.

```
/**
 @return void
 */
function test()
{
    return true;
}
```

**Comment by kukulich** [comment]
The behaviour is expected. The sniff does not detect specific type hints based on `return` statements. It can only recognize "returns something/nothing". `void` is bad here so the generated native type hint is `null`.

**Comment by othercorey** [comment]
But `null` is not a valid return type. Why wouldn't it just skip?

https://3v4l.org/E6L2U#v8.0.8

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/59bdaf85630935222f94f3950ec3f3751c352d23
